### ISSUE TITLE
Bug: Tree view: collapse only affects highest level

### ISSUE THREAD

**Issue by andresgongora** [SkepticMystic/breadcrumbs#555]
**Describe the bug**

In _Tree View_, clicking on the _Collapse_ button only collapses the highest (first) level. Initially it looks fine, but clicking on any on the (now collapsed) top level elements will expand the whole sub tree instead of only showing the second level.

**To Reproduce**

1. Have a wiki with multiple breadcrumbs forming a complex tree structure.
2. Navigate to one of the top-most pages, open the Tree View for it.
3. In the Tree View pane, click on _Collapse_; all elements should collapse.
4. Expand any element that has multiple recursively nested elements.

**Expected behavior**
All elements should me nested and would need to be expanded one by one.

**Screenshots**

1. After collapsing all.  
 ![image](https://github.com/SkepticMystic/breadcrumbs/assets/12051510/3e466766-9769-448f-b2ed-3b0cdcdf0e69)
2. After expanding one of the top level elements, curret behavior.  
![image](https://github.com/SkepticMystic/breadcrumbs/assets/12051510/728021fa-df8c-4698-8e34-3184bfcb22df)
3. Expected behavior.
![image](https://github.com/SkepticMystic/breadcrumbs/assets/12051510/4da792c3-c6b5-4643-aa62-d7516e1905da)

**Desktop (please complete the following information):**

-   OS: Manjaro
-   Breadcrumbs version: 4.2.35
-   Obsidian version: 1.5.12-1

**Additional context**
I've assumed this is a bug since the normal _collapse_ behavior on most other tools I use on a daily basis is to "fully" collapse everything recursively. If this is not a bug but a feature (maybe to make it faster?), would it be possible to add an option to toggle the behavior instead?

**Comment by andresgongora** [comment]
Might this be a duplicate of #546 ? The other one is about codeblocks, but it might have the same root cause under the hood

**Comment by ybalkind** [comment]
> Might this be a duplicate of #546 ? The other one is about codeblocks, but it might have the same root cause under the hood

Yup its describing the same behaviour but referring to a different view. #546 is still quite an issue for me, hope it gets looked at.

**Comment by GardarikanetS** [comment]
+1, It worries me

**Comment by michaelpporter** [comment]
@andresgongora This is working in the latest version of the plugin and Obsidian.
